protocol/0085-RVST-rewards_vesting.md

```diff
@@ -0,0 +1,50 @@
+# Rewards Vesting Specification
+
+## Summary
+
+The aim of the rewards vesting mechanics is to prevent farming rewards by delaying the payout of rewards through vesting. To encourage longer term behaviour parties can accelerate their rewards vesting rate through the [activity streak program](./0086-ASPR-activity_streak_program.md).
+
+## Network Parameters
+
+- `rewards.vesting.baseRate`: the proportion of rewards in a vesting account which are vested each epoch
+- `rewards.vesting.minimumTransfer`: the minimum amount (expressed in quantum) which can be vested each epoch
+
+## Vesting mechanics
+
+As detailed in [distributing rewards](), each party has their rewards paid into vesting rewards accounts (one for each asset).
+
+At the end of each epoch, a proportion of the rewards accumulated in each "vesting" account should be released and transferred to the respective "vested" account. The percentage released can be scaled by the account owner increasing their [activity streak](./0086-ASPR-activity_streak_program.md) and a minimum transfer amount will be applied to ensure the account is eventually emptied. The proportion released and minimum applied are controlled for parameters for the asset.
+
+Now, let:
+
+- $T$ be the amount to be "vested" (transferred from the vesting account to the vested account)
+- $B_{vested}$ be the total quantum amount in the vesting account
+- $r_{bvr}$ be the network parameter `rewards.vesting.baseRate`
+- $a$ be the account owners current [`activity_streak_vesting_multiplier`](./0086-ASPR-activity_streak_program.md#setting-activity-benefits)
+- $m_{mva}$ be the network parameter `rewards.vesting.minimumTransfer`
+
+The quantum amount to be transferred from each "vesting" account to the relevant "vested" account is defined as:
+$$
+T = min(B_{vested} * r * a, m)
+$$
+
+When transferring funds from the vesting account to the vested account, a new transfer type should be used, `TRANSFER_TYPE_REWARDS_VESTED`.
+
+## APIs
+
+### Accounts API
+
+Must expose the following:
+
+- every account with `ACCOUNT_TYPE_VESTING_REWARDS` for each party
+- every account with `ACCOUNT_TYPE_VESTED_REWARDS` for each party
```

Comment:
The initial document mentioned:
```
Rewards are paid into a (new) locked (unvested) rewards account N epochs after they are earned (this is set on reward distribution, and can be different for each reward. 
There’s a network param for the maximum value of N.
```

That seems to translate better into 3 accounts.
- A first locked one where the funds are transfered from the reward pool onto, and locked for the given party
- A second one where the funds are vesting.
- An a last one were they are transfered once vested.

Reply:
hum actually 2 accounts would work, we can just keep track internally what amount is available for being distributed, and what amount is still locked before starting vesting.

Reply:
Think I agree 3 accounts would actually be clearer to explain in docs and expose through APIs. If it is not a big deal implementation wise could we go with that?

If you want to stick with two, I would definitely rename the `ACCOUNT_TYPE_VESTING_REWARDS` to `ACCOUNT_TYPE_UNVESTED_REWARDS` to make it clear some funds in it are locked before they start to vest.

Reply:
```
If you want to stick with two, I would definitely rename the ACCOUNT_TYPE_VESTING_REWARDS to ACCOUNT_TYPE_UNVESTED_REWARDS to make it clear some funds in it are locked before they start to vest.
```
Yea can do that.